python/python/tests/test_fragment.py

```diff
@@ -72,3 +72,26 @@ def test_write_fragment_with_progress(tmp_path: Path):
     LanceFragment.create(tmp_path, df, progress=progress)
     assert progress.begin_called == 1
     assert progress.complete_called == 1
+
+
+def test_dataset_progress(tmp_path: Path):
```

Comment:
There were several bugs with `FileSystemFragmenWriteProgress`, so added a unit test for it.